src/pdf-proxy.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * This file is part of Serlo.org Cloudflare Worker.
+ *
+ * Copyright (c) 2021 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2021 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/serlo.org-cloudflare-worker for the canonical source repository
+ */
+
+import { isInstance, Url, SentryFactory } from './utils'
+
+export async function pdfProxy(
+  request: Request,
+  sentryFactory: SentryFactory
+): Promise<Response | null> {
+  const url = Url.fromRequest(request)
+
+  if (!isInstance(url.subdomain)) return null
+  const pdfMatch = /^\/api\/pdf\/(\d+)/.exec(url.pathname)
+  if (!pdfMatch) return null
+
+  const sentry = sentryFactory.createReporter('pdf-proxy')
+
+  try {
+    url.hostname = 'pdf.serlo.org'
+    url.pathname = `/api/${pdfMatch[1]}`
+
+    const pdfRes = await fetch(url.href, {
+      cf: { cacheTtl: 24 * 60 * 60 },
+    } as unknown as RequestInit)
+
+    if (pdfRes.ok) return pdfRes
+  } catch (e) {
```

Comment:
There is no possibility for an exception to be thrown -> delete it (maybe we want to add a global try-catch to handle uncatched exceptions)

Reply:
Do you want to report for non 200 responses?!

Reply:
tbh. it's just copy&paste from the `quickbar-proxy`.
Probably same case there. I guess 500 or 404 are unlikely but worth reporting.

Reply:
Yeah, you can also report any other response and narrow it down later when you false positives in sentry...